### ISSUE TITLE
Consistent exception handling

### ISSUE THREAD

**Issue by jenetics** [jenetics/jpx#87]
The exceptions, thrown in the case of a parsing error, should be fit with the already thrown exception. In the case of an un-parsable value (e.g. 'asf' for the elevation), an `IllegalArgumentException` is thrown. This exception is neither documented, nor a good fit into the existing exception strategy.

Use `java.io.InvalidObjectException` for not-parsable values.

**Comment by jenetics** [comment]
An `IOE` should be thrown only by the `GPX.Reader.read` methods and not when the GPX object are created programmatically.

The following code should throw and `InvalidObjectException` in the case of invalid input values.
```java
final GPX gpx = GPX.read("./track.gpx");
```

In contrast this code should still throw an `IllegalArgumentException`.
```java
  final GPX gpx = GPX.builder()
      .addTrack(track -> track
          .addSegment(segment -> segment
              .addPoint(p -> p.lat(10000).lon(16.3738189).ele(162))))
      .build();
```

**Comment by Segelzwerg** [comment]
Oh yes that makes absolutely sense. If would like, I can look into it tomorrow and update my PR.

**Comment by jenetics** [comment]
> Oh yes that makes absolutely sense. If would like, I can look into it tomorrow and update my PR.

OK, would be great.

**Comment by Segelzwerg** [comment]
I have follow up question. Currently the `GPX.Reader.read` does not throw `IllegalArgumentException` only the `GPX.Reader.fromString`. So the question is you want `GPX.Reader.read` to throw an additional `InvalidObjectException`?

**Comment by jenetics** [comment]
The `IAE` thrown by the `GPX.Reader.read` is not document, but can be thrown. This is the inconsistency this ticket should fix. The exception handling of the `GPX.Reder.fromString` is OK, because if the string doesn't describe valid GPX data, its an _illegal argument_.
So the `GPX.Read.read` methods should throw an `IOE` for GPX data where the `GPX.Reader.fromString` method would throw an `IAE`. Makes sense?

Could you base the fix on the r1.6.0 branch and create a new PR into this branch? I would like to have this fix in the 1.6.0 release. 
Thank you!

**Comment by jenetics** [comment]
Merged into r1.6.0 branch.
